### DISCUSSION TITLE
High Available 2 member (active-passive) ETCD configuration possible??

### DISCUSSION THREAD

**Discussion by sanamsarath** [etcd-io/etcd#15481] [Q&A]
HI,

We use ETCD in our product and we got a hard requirement from the customer to support High Available solution on a 2 node Hardware. Wondering if it is possible to setup ETCD members in an active-passive mode and make the passive member to transition to leader in case of active member failure?? The trigger to transition can be external event like manual as well if required (we can work on automating the steps using a side car)

I looked couple of options like learner member, mirror maker but not able to figure out if they can help to our requirement. 

Appreciate any kind of suggestions or help.

Thanks!!

**Comment by chaochn47** [comment]
No. A 2 member cluster does not tolerate hardware failures in one of the member. 

> Make learner standby failover node

https://etcd.io/docs/v3.5/learning/design-learner/#proposed-features-for-future-releases

I think this feature is not supported, tracked or prioritized if I understand your question properly. Not to mention promoting learner requires quorum which seems not working if the only voting member in the cluster failed. 

However, if you do want to use etcd to overcome this constraint, I would suggest build another layer on top of etcd to satisfy the active-passive requirement.

**Comment by serathius** [comment]
Answer is no, Raft consensus protocol requires uneven number of members to achieve quorum.
